lib/yerror.js

```diff
@@ -0,0 +1,11 @@
+const util = require('util')
```

Comment:
@satazor @iarna hadn't thought of the front-end use-case; I know in the past a couple people had used something like browserify to load yargs in the browser. It might be worth switching from `util.inherits` to something like:

```js
MyError.prototype = Object.create(Error.prototype);
MyError.prototype.constructor = MyError;
```

Which I lifted rom the MDN docs.

@satazor to answer your question; I get a lot of pressure from the user-base to keep yargs to as few dependencies as possible:

https://twitter.com/guillaumeQD/status/807917750425440256

More so than a lot of libraries I work on, this has motivated me to opt for less dependencies where possible -- I think people are extra sensitive about library-size, because yargs is something you tack onto a project to add CLI functionality, it's not seen as a core part of the application logic.